Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,215 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
+
+        private const string DEFAULT_MESSAGE = "N/A.";
+
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return DEFAULT_MESSAGE;
+
+            var highestHistoryBalance = peopleAndBalances.Max(GetLastHistoryBalance);
+
+            var names = peopleAndBalances.GetNamesForSameBalance(highestHistoryBalance, GetLastHistoryBalance);
+
+            return $"{FormatPersonName(names)} had the most money ever. {FormatBalance(highestHistoryBalance)}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return DEFAULT_MESSAGE;
+
+            var peopleLossBalance = new string[peopleAndBalances.Length];
```

Comment:
Both names are similar and confusing. peopleLossBlanace sounds like a single thing yet this is an array.